### PR TITLE
Add cached conda env file for Buck CI workflow

### PR DESCRIPTION
**Pull request by PratsBhatt**
Fixes - T137631262

Caching conda dependencies for build workflows.
Conda dependencies have been gathered from the workflow https://github.com/pytorch/pytorch/blob/master/.github/workflows/_buck-build-test.yml

The pull request updates the action from `conda-incubator/setup-miniconda@v2` to `pytorch/test-infra/.github/actions/setup-miniconda@main` as it supports caching.

Test Plan:

Running the `ciflow/periodic` which runs the ci builds `buck-build-test` workflow. Expected output is to have all the conda dependencies cached.

Screenshot 2022-11-22 at 15 44 20

### CODE REVIEW

File: .github/workflows/_buck-build-test.yml [hunk 1]
```diff
@@ -21,11 +21,10 @@ jobs:
...
       - name: Setup miniconda
-        uses: conda-incubator/setup-miniconda@v2
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
         with:
-          auto-update-conda: true
           python-version: 3.8
-          activate-environment: build
+          environment-file: .github/requirements/conda-env-ubuntu-X64
 
       - name: Install dependencies
```
Comment (huydhn) [inline .github/workflows/_buck-build-test.yml:29]: You would need to remove this step `Install dependencies` to test the change.  The idea is that this step won't be needed anymore

File: .github/workflows/_buck-build-test.yml [hunk 2]
```diff
@@ -21,11 +21,10 @@ jobs:
           distribution: 'temurin'
 
       - name: Setup miniconda
-        uses: conda-incubator/setup-miniconda@v2
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
         with:
-          auto-update-conda: true
           python-version: 3.8
-          activate-environment: build
+          environment-file: .github/requirements/conda-env-ubuntu-X64
```
Comment (huydhn) [inline .github/workflows/_buck-build-test.yml:27]: I think it's better to use the syntax `.github/requirements/conda-env-${{ runner.os }}-${{ runner.arch }}` here.  From what I see in https://github.com/pytorch/pytorch/actions/runs/3516771558/jobs/5893753273. The file name would become  `conda-env-Linux-X64` instead

File: .github/workflows/_buck-build-test.yml (hunk 1 above)
Comment (PratsBhatt) [reply .github/workflows/_buck-build-test.yml]: Yes, I have removed the block.

File: .github/workflows/_buck-build-test.yml (hunk 2 above)
Comment (PratsBhatt) [reply .github/workflows/_buck-build-test.yml]: Implemented the changes.

**Comment by PratsBhatt** [comment]
One of the workflow is failing and it is due to CUDA OOM. It doesn't seem to be related to this PR.
Screenshot 2022-11-22 at 10 26 51

**Review by huydhn — APPROVED** [review]
LGTM!

**Pull request closed without merging**